src/Parser/Context.hs

```diff
@@ -353,12 +355,27 @@ handleContextBlock needsSep  tok@(Token tk tp te) c =
       , isTopTok tc
       -> pure ( Token TcVBegin tp tp `Result` Working tok
               , CtxEmptyBlock (Just TcVEnd) : CtxInstBody : ck )
-    -- Offside rule for empty instancees
+
+    -- Offside rule for empty instances
     (_, CtxInstHead offside:ck)
       | spCol tp <= spCol offside
       -> pure ( Token TcVBegin tp tp `Result` Token TcVEnd tp tp `Result` Working tok
               , ck)
 
+    -- @type function ... begin@ ~~> Replace type head with body context
+    (TcBegin, CtxTypeHead offside:ck)
+      | spCol tp >= spCol offside
+      -> pure (Result tok Done, CtxEmptyBlock Nothing : CtxTypeFunBody : CtxBracket TcEnd : ck)
+
+    (tc, CtxTypeHead cls:ck)
+      | spCol tp > spCol cls
+      , isTopTok tc && tc /= TcType
```

Comment:
Why the `tc /= TcType`?

Reply:
In fact, why this rule at all? It's not clear when it will be triggered given the syntax for type functions.

Reply:
idk i don't know this stuff lol it just didn't work without the tc /= TcType